File: library/src/main/java/org/crispy_fish/filetype/staging/StagingLineReader.java
package org.crispy_fish.filetype.staging;

public interface StagingLineReader<L> {

    String getRegisteredDriverClass(L line);

    String getRegisteredDriverNumber(L line);

    String getRegisteredDriverName(L line);

    String getRegisteredDriverCar(L line);

    String getRegisteredDriverCarColor(L line);

    String getRunNumber(L line);

    String getRunRawTime(L line);

    String getRunPaxTime(L line);

    String getRunPenalty(L line);

    String getRunTimestamp(L line);

    String getDriverPositionInClassForDay(L line);

    String getDriverBestTimeOfDay(L line);
}

File: library/src/main/kotlin/tech/coner/crispyfish/internal/filetype/staging/StagingLineReader.kt
package tech.coner.crispyfish.internal.filetype.staging

internal interface StagingLineReader<L> {

    fun getRegisteredDriverClass(line: L): String?

    fun getRegisteredDriverNumber(line: L): String?

    fun getRegisteredDriverName(line: L): String?

    fun getRegisteredDriverCar(line: L): String?

    fun getRegisteredDriverCarColor(line: L): String?

    fun getRunNumber(line: L): String?

    fun getRunRawTime(line: L): String?

    fun getRunPaxTime(line: L): String?

    fun getRunPenalty(line: L): String?

    fun getRunTimestamp(line: L): String?

    fun getDriverPositionInClassForDay(line: L): String?

    fun getDriverBestTimeOfDay(line: L): String?
}
